src/server.py

```diff
@@ -18,65 +18,85 @@
-    args = ",".join(parsed_parameters)
-
-    async def run_subprocess():
-        try:
-            result = subprocess.run(
-                ["python3", "src/cli.py", args],
-                capture_output=True,
-                text=True,
-                check=True,
-            )
-            return result.stdout
-        except subprocess.CalledProcessError as e:
-            # Print the error message from the subprocess
-            print("Error message from subprocess:", e.stderr)
-            # Raise the error again to propagate it
-            raise e
-
-    # Run subprocess asynchronously
-    loop = asyncio.new_event_loop()
-    asyncio.set_event_loop(loop)
-    result = loop.run_until_complete(run_subprocess())
-    return result
+def create_app():
+    """
+    Application factory function
+    """
+
+    # Load environment variables from .env file
+    env = ServerSettings()
+
+    app = Flask(__name__)
+    CORS(app)
+
+    @app.route("/help")
+    def serve_help():
+        """
+        Servers the help.txt file
+        """
+        return send_from_directory(
+            Path(__file__).resolve().parents[1], "help.txt"
+        )
+
+    @app.route("/home")
+    def serve_index():
+        """
+        Servers index.html
+        """
+        return render_template("index.html", env_vars=env.model_dump())
+
+    @app.route("/script.js")
+    def serve_script():
+        """
+        Servers javascript
+        """
+        return send_file("static/script.js")
+
+    @app.route("/")
+    def default_route():
+        """
+        Default route, serves surf report
+        """
+        query_parameters = urllib.parse.parse_qsl(
+            request.query_string.decode(), keep_blank_values=True
+        )
+        parsed_parameters = []
+
+        for key, value in query_parameters:
+            if value:
+                parsed_parameters.append(f"{key}={value}")
+            else:
+                parsed_parameters.append(key)
+
+        # Join the parsed parameters list into a single string
+        args = ",".join(parsed_parameters)
+
+        async def run_subprocess():
+            try:
+                result = subprocess.run(
+                    ["python3", "src/cli.py", args],
+                    capture_output=True,
+                    text=True,
+                    check=True,
+                )
+                return result.stdout
+            except subprocess.CalledProcessError as e:
+                # Print the error message from the subprocess
+                print("Error message from subprocess:", e.stderr)
+                # Raise the error again to propagate it
+                raise e
+
+        # Run subprocess asynchronously
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        result = loop.run_until_complete(run_subprocess())
+        return result
+
+    return app
 
 
 if __name__ == "__main__":
+    app = create_app()
+    env = ServerSettings()
```

Comment:
```suggestion
    env = ServerSettings()
    app = create_app(env)
```

By passing the env parameter to the `create_app()` function, we can inject environment variable values from external sources. This approach enhances the flexibility of our tests, allowing us to easily modify and test different environment configurations!